### ISSUE TITLE
Add support for more complex syntax to compare two jsonb field references.

### ISSUE THREAD

**Issue by elhigu** [Vincit/objection.js#18]
Currently only case where we are able to do something like 

```
where ("jsonCol"->'a')::jsonb = ("jsonCo"->.'b')::jsonb
```

is with equality, subset and superset operators. This case was straight forward to implement, because all those operands works nicely with `jsonb` types.

To be able to use all other operators we would need to tell explicitly casting how `jsonb` fields are interpreted for given operator.

e.g. 

```
select * from "ModelJson" where ("ModelJson"."jsonObject"#>'{a}')::text LIKE ("ModelJson"."jsonObject"#>'{b}')::text;
```

Would require that we actually have to give type casting for both sides of operand. That is due to nature of jsonb fields where PostgreSQL just doesn't know how types should be casted, like the case is with normal columns.

One way to support type casting could be to extend filed expression a bit more to allow to tell to what type reference should be casted e.g.

``` javascript
.whereJson("ModelJson.jsonObject:a.0<numeric>", ">", "ModelJson.jsonObject:b.0<numeric>")
```

would yield 

```
select * from "ModelJson" where ("ModelJson"."jsonObject"#>>'{a,0}')::numeric > ("ModelJson"."jsonObject"#>>'{b,0}')::numeric;
```

notice that when type casting is given, field value is extracted with `#>>` operator instead of `#>` to be able to the given casting.

**Comment by koskimas** [comment]
How about the basic C/Java cast syntax:

```js
.whereJson("(numeric) ModelJson.jsonObject:a.0", ">", "(numeric) ModelJson.jsonObject:b.0")
```

**Comment by elhigu** [comment]
It should would work too. 

Good thing is that putting type to front is that probably there is less chance that table name is having e.g. `(numeric)` as prefix in its name than json key of having `<numeric>` postfix. So to put cast to front is probably safer choice. 

Just thinking of syntax I would have liked to put it after reference so that it is located near to actual field it refers. I actually never liked to use type cast for struct field in C without parenthesis... I usually did it like this `(int)(imStruct.uintfield)` :)

So syntax is fine either way for me, but is this only way how to solve this casting problem? Any other ideas?

**Comment by koskimas** [comment]
Your cast syntax if good also. I just threw in an idea. In most cases we could determine the type from json schema, couldnt we? And if we can't we could throw an informative error telling the user to use an explicit cast.

**Comment by elhigu** [comment]
Now that I have thought this further... 

If we are going to write that cref() object to allow putting column references easily anywhere, we could actually give casting as a parameter for that object. So the same casting code would work with all kind of column references, not only with JSON stuff.

**Comment by elhigu** [comment]
It seems that also this one will be implemented as part of #16 with a bit different syntax... e.g. `ref('Table.Column:jsonAttr').asFloat()`

**Comment by elhigu** [comment]
Implemented in https://github.com/Vincit/objection.js/pull/270 now ref() syntax supports also casting and .as('someOtherName') modifiers.
